File: Core/CommandParser.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Uber.Messages;
using PluginInterface;

namespace Uber.Core
{
    class CommandParser
    {
        public static void Parse(string Input)
        {
            string[] Params = Input.Split(' ');

            switch (Params[0])
            {
                case "reload_models":

                    UberEnvironment.GetGame().GetRoomManager().LoadModels();
                    break;

                case "reload_bans":

                    UberEnvironment.GetGame().GetBanManager().LoadBans();
                    break;

                case "reload_navigator":

                    UberEnvironment.GetGame().GetNavigator().Initialize();
                    UberEnvironment.GetLogging().WriteLine("Re-initialized navigator successfully.");

                    break;

                case "reload_items":

                    UberEnvironment.GetGame().GetItemManager().LoadItems();
                    UberEnvironment.GetLogging().WriteLine("Please note that changes may not be reflected immediatly in currently loaded rooms.");

                    break;

                case "reload_help":

                    UberEnvironment.GetGame().GetHelpTool().LoadCategories();
                    UberEnvironment.GetGame().GetHelpTool().LoadTopics();
                    UberEnvironment.GetLogging().WriteLine("Reloaded help categories and topics successfully.");

                    break;

                case "reload_catalog":

                    UberEnvironment.GetGame().GetCatalog().Initialize();
                    UberEnvironment.GetGame().GetClientManager().BroadcastMessage(new ServerMessage(441));
                    UberEnvironment.GetLogging().WriteLine("Published catalog successfully.");

                    break;

                case "reload_roles":

                    UberEnvironment.GetGame().GetRoleManager().LoadRoles();
                    UberEnvironment.GetGame().GetRoleManager().LoadRights();
                    UberEnvironment.GetLogging().WriteLine("Reloaded ranks and rights successfully.");

                    break;

                case "plugins":

                    UberEnvironment.GetLogging().WriteLine("The following plugins are currently loaded:");

                    lock (UberEnvironment.GetPluginHandler().AvailablePlugins)
                    {
                        foreach (Plugins.Types.AvailablePlugin Plugin in UberEnvironment.GetPluginHandler().AvailablePlugins)
                        {
                            UberEnvironment.GetLogging().WriteLine("-------------------------------------------------------------");
                            UberEnvironment.GetLogging().WriteLine("Name: " + Plugin.Instance.Name);
                            UberEnvironment.GetLogging().WriteLine(Plugin.Instance.Description);
                            UberEnvironment.GetLogging().WriteLine("Version: " + Plugin.Instance.Version);
                            UberEnvironment.GetLogging().WriteLine("Author: " + Plugin.Instance.Author);
                            UberEnvironment.GetLogging().WriteLine("-------------------------------------------------------------");
                        }
                    }

                    break;

                case "unload_all_plugins":

                    UberEnvironment.GetPluginHandler().UnloadPlugins();
                    UberEnvironment.GetLogging().WriteLine("All plugins have been unloaded.");

                    break;

                case "unload_plugin":

                    if (UberEnvironment.GetPluginHandler().UnloadPlugin(MergeParams(Params, 1)))
                    {
                        UberEnvironment.GetLogging().WriteLine("Plugin unloaded successfully.");
                        UberEnvironment.GetLogging().WriteLine("Take note that a plugin may still be running processes even when unloaded. To fully disable a plugin, remove the .dll file from the directory and restart the server.", LogLevel.Warning);
                    }
                    else
                    {
                        UberEnvironment.GetLogging().WriteLine("Plugin could not be unloaded. You probably mis-typed the plugin name: " + MergeParams(Params, 1), LogLevel.Warning);
                    }

                    break;

                case "cls":

                    UberEnvironment.GetLogging().Clear();
                    break;

                case "help":

                    UberEnvironment.GetLogging().WriteLine("Available commands are: cls, close, help, reload_catalog, reload_navigator, reload_roles, reload_help, reload_items, plugins, unload_all_plugins, unload_plugin [name]");

                    break;

                case "close":

                    UberEnvironment.Destroy();

                    break;

                default:

                    UberEnvironment.GetLogging().WriteLine("Unrecognized command or operation: " + Input + ". Use 'help' for a list of available commands.", LogLevel.Warning);

                    break;
            }
        }

        public static string MergeParams(string[] Params, int Start)
        {
            StringBuilder MergedParams = new StringBuilder();

            for (int i = 0; i < Params.Length; i++)
            {
                if (i < Start)
                {
                    continue;
                }

                if (i > Start)
                {
                    MergedParams.Append(" ");
                }

                MergedParams.Append(Params[i]);
            }

            return MergedParams.ToString();
        }
    }
}

File: src/main/java/com/uber/server/core/CommandParser.java
package com.uber.server.core;

import com.uber.server.game.GameEnvironment;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * Command parser for console commands.
 * Ported from Core/CommandParser.cs
 */
public class CommandParser {
    private static final Logger logger = LoggerFactory.getLogger(CommandParser.class);
    
    /**
     * Parses and executes a console command.
     * @param input Command input string
     * @param environment GameEnvironment instance
     */
    public static void parse(String input, GameEnvironment environment) {
        if (input == null || input.trim().isEmpty()) {
            return;
        }
        
        String[] params = input.trim().split("\\s+");
        if (params.length == 0) {
            return;
        }
        
        String command = params[0].toLowerCase();
        
        switch (command) {
            case "reload_models":
                environment.getGame().getRoomManager().loadModels();
                logger.info("Reloaded room models successfully.");
                break;
                
            case "reload_bans":
                environment.getGame().getBanManager().loadBans();
                logger.info("Reloaded bans successfully.");
                break;
                
            case "reload_navigator":
                environment.getGame().getNavigator().initialize();
                logger.info("Re-initialized navigator successfully.");
                break;
                
            case "reload_items":
                environment.getGame().getItemManager().loadItems();
                logger.info("Reloaded items successfully. Please note that changes may not be reflected immediately in currently loaded rooms.");
                break;
                
            case "reload_help":
                environment.getGame().getHelpTool().loadCategories();
                environment.getGame().getHelpTool().loadTopics();
                logger.info("Reloaded help categories and topics successfully.");
                break;
                
            case "reload_catalog":
                environment.getGame().getCatalog().initialize();
                // Broadcast catalog update to all clients
                com.uber.server.messages.ServerMessage catalogUpdate = new com.uber.server.messages.ServerMessage(441);
                environment.getGame().getClientManager().broadcastMessage(catalogUpdate);
                logger.info("Published catalog successfully.");
                break;
                
            case "reload_roles":
                environment.getGame().getRoleManager().loadRoles();
                environment.getGame().getRoleManager().loadRights();
                logger.info("Reloaded ranks and rights successfully.");
                break;
                
            case "plugins":
                // TODO: Implement when PluginHandler is ported
                logger.info("The following plugins are currently loaded:");
                logger.info("(Plugin system not yet implemented)");
                break;
                
            case "unload_all_plugins":
                // TODO: Implement when PluginHandler is ported
                logger.info("All plugins have been unloaded.");
                break;
                
            case "unload_plugin":
                if (params.length < 2) {
                    logger.warn("Usage: unload_plugin <name>");
                    break;
                }
                String pluginName = mergeParams(params, 1);
                // TODO: Implement when PluginHandler is ported
                logger.info("Plugin unloaded successfully.");
                logger.warn("Take note that a plugin may still be running processes even when unloaded.");
                break;
                
            case "cls":
                // Clear console (not really possible in Java, but log a message)
                logger.info("--- Console cleared ---");
                break;
                
            case "help":
                logger.info("Available commands are: cls, close, help, reload_catalog, reload_navigator, reload_roles, reload_help, reload_items, plugins, unload_all_plugins, unload_plugin [name]");
                break;
                
            case "close":
            case "quit":
            case "exit":
                logger.info("Shutting down server...");
                environment.destroy();
                System.exit(0);
                break;
                
            default:
                logger.warn("Unrecognized command or operation: {}. Use 'help' for a list of available commands.", input);
                break;
        }
    }
    
    /**
     * Merges parameters from a specific index onwards.
     * @param params Parameter array
     * @param start Start index
     * @return Merged string
     */
    private static String mergeParams(String[] params, int start) {
        if (start >= params.length) {
            return "";
        }
        
        StringBuilder merged = new StringBuilder();
        for (int i = start; i < params.length; i++) {
            if (i > start) {
                merged.append(" ");
            }
            merged.append(params[i]);
        }
        return merged.toString();
    }
}
